src/library.h

```diff
@@ -51,6 +52,7 @@ public slots:
     void bookmarksChanged();
 
 private:
+    QMutex mutex;
     kiwix::Library m_library;
     QString m_libraryDirectory;
     QMap<QString, QStringList> m_monitorDirZims;
```

Comment:
I wonder (not sure) if this should be renamed to `m_knownZimsInDir`

Reply:
Yeah, these all naming makes sense because it's a generic function now. Not only monitorDir specific